llvm/lib/CodeGen/MachineBasicBlock.cpp

```diff
@@ -1097,6 +1097,31 @@ static bool jumpTableHasOtherUses(const MachineFunction &MF,
   return false;
 }
 
+class MBBSplitCriticalEdgeDelegate : public MachineFunction::Delegate {
```

Comment:
Call this `SlotIndexUpdateDelegate` or similar? (as it doesn't seem very specific to splitting critical edges now.